esm_runscripts/slurm.py

```diff
@@ -80,16 +82,40 @@ def mini_calc_reqs(config, model, hostfile, start_proc, end_proc):
                     end_proc += config[model]["nprocar"] * config[model]["nprocbr"]
 
         else:
-            return start_proc, end_proc
+            return start_proc, start_core, end_proc, end_core
+
+        scriptfolder = config["general"]["thisrun_scripts_dir"] + "../work/"
+        if config["general"].get("taskset", False): 
+            command = "./" + config[model]["execution_command_script"]
+            scriptname="script_"+model+".ksh"
+            with open(scriptfolder+scriptname, "w") as f:
+                f.write("#!/bin/ksh"+"\n")
+                f.write("export OMP_NUM_THREADS="+str(config[model]["omp_num_threads"])+"\n")
+                f.write(command+"\n")
+            os.chmod(scriptfolder+scriptname, 0o755)
+
+            progname="prog_"+model+".sh"
+            with open(scriptfolder+progname, "w") as f:
+                f.write("#!/bin/sh"+"\n")
+                f.write("(( init = "+str(start_core)+" + $1 ))"+"\n")
+                f.write("(( index = init * "+str(config[model]["omp_num_threads"])+" ))"+"\n")
+                f.write("(( slot = index % "+str(config["computer"]["cores_per_node"])+" ))"+"\n")
+                f.write("echo "+model+" taskset -c $slot-$((slot + "+str(config[model]["omp_num_threads"])+" - 1"+"))"+"\n")
+                f.write("taskset -c $slot-$((slot + "+str(config[model]["omp_num_threads"])+" - 1)) ./script_"+model+".ksh"+"\n")
+            os.chmod(scriptfolder+progname, 0o755)
+
         if "execution_command" in config[model]:
             command = "./" + config[model]["execution_command"]
         elif "executable" in config[model]:
             command = "./" + config[model]["executable"]
         else:
-            return start_proc, end_proc
-        hostfile.write(str(start_proc) + "-" + str(end_proc) + "  " + command + "\n")
-        start_proc = end_proc + 1
-        return start_proc, end_proc
+            return start_proc, start_core, end_proc, end_core
+
+        with open(path, "a") as hostfile:
+            hostfile.write(str(start_proc) + "-" + str(end_proc) + "  " + command + "\n")
+            start_proc = end_proc + 1
+            start_core = end_core + 1
+        return start_proc, start_core, end_proc, end_core
```

Comment:
As mentioned in the previous comment, I would rather see this done outside of this function.